beacon-chain/node/node.go

```diff
@@ -191,20 +159,18 @@ func New(cliCtx *cli.Context, cancel context.CancelFunc, opts ...Option) (*Beaco
 		slasherBlockHeadersFeed: new(event.Feed),
 		slasherAttestationsFeed: new(event.Feed),
 		serviceFlagOpts:         &serviceFlagOpts{},
+		forkChoicer:             doublylinkedtree.New(),
+		clockWaiter:             synchronizer,
+		initialSyncComplete:     make(chan struct{}),
+		syncChecker:             &initialsync.SyncChecker{},
 	}
 
-	beacon.initialSyncComplete = make(chan struct{})
-	beacon.syncChecker = &initialsync.SyncChecker{}
 	for _, opt := range opts {
 		if err := opt(beacon); err != nil {
 			return nil, err
 		}
 	}
 
-	synchronizer := startup.NewClockSynchronizer()
```

Comment:
Its important not to change the ordering here, if you notice this is the setup after the beacon options are initialized above. Your PR changes it to before.